app/src/main/java/org/simple/clinic/summary/bloodsugar/view/BloodSugarSummaryView.kt

```diff
@@ -147,7 +157,23 @@ class BloodSugarSummaryView(
   }
 
   override fun showBloodSugarTypeSelector() {
-    activity.startActivity(Intent(context, BloodSugarTypePickerSheet::class.java))
+    activity.startActivityForResult(BloodSugarTypePickerSheet.intent(context), TYPE_PICKER_SHEET)
+  }
+
+  private fun openEntrySheetAfterTypeIsSelected(onDestroys: Observable<ScreenDestroyed>) {
+    screenRouter.streamScreenResults()
+        .ofType<ActivityResult>()
+        .filter { it.requestCode == TYPE_PICKER_SHEET && it.succeeded() }
+        .takeUntil(onDestroys)
+        .map { showBloodSugarEntrySheet(it) }
+        .subscribe()
+  }
+
+  private fun showBloodSugarEntrySheet(it: ActivityResult) {
```

Comment:
I would change the parameter of this method to accept the blood sugar measurement type directly, and then extract this out at the method call site. Otherwise, this method now knows the implementation detail that the type is selected via the `BloodSugarTypePickerSheet`.